contracts/core/src/octane/Staking.sol

```diff
@@ -299,6 +321,18 @@ contract Staking is OwnableUpgradeable, EIP712Upgradeable {
         emit Delegate(delegator, validator, msg.value);
     }
 
+    /**
+     * @notice Undelegate tokens from a validator
+     * @param delegator The address of the delegator
+     * @param validator The address of the validator to undelegate from
+     */
+    function _undelegate(address delegator, address validator) internal {
+        require(!isAllowlistEnabled || isAllowedValidator[validator], "Staking: not allowed val");
```

Comment:
Not sure if this check makes sense, maybe I should remove it?

Reply:
we need to charge a fee to avoid spam attacks